File: JAVA/3042. Count Prefix and Suffix Pairs I.java
class Solution {
    public int countPrefixSuffixPairs(String[] w) {
        int c=0;
        for(int i=0;i<w.length;i++){
            for(int j=i+1;j<w.length;j++){
                if(isPrefixAndSuffix(w[i],w[j]))
                c++;
            }
        }
        return c;
    }
    private boolean isPrefixAndSuffix(String s1,String s2){
        if(s1.length()>s2.length())
        return false;
        if(s2.substring(0,s1.length()).equals(s1)&&s2.substring(s2.length()-s1.length()).equals(s1))
        return true;
        return false;
    }
}

File: C++/3042. Count Prefix and Suffix Pairs I.cpp
class Solution {
public:
    bool isPrefixAndSuffix(string s1,string s2){
        if(s1.size()>s2.size())
        return false;
        if(s2.substr(0,s1.size())==s1 && s2.substr(s2.size()-s1.size(),s1.size())==s1)
        return true;
        return false;
    }
    int countPrefixSuffixPairs(vector<string>& w) {
        int c=0;
        for(int j=0;j<w.size();j++){
        for(int i=j+1;i<w.size();i++){
            if(isPrefixAndSuffix(w[j],w[i]))
            c++;
        }}
        return c;
    }
};